pyomo/contrib/pyros/uncertainty_sets.py

```diff
@@ -570,6 +589,34 @@ def dim(self):
     def geometry(self):
         return Geometry.LINEAR
 
+    @property
+    def parameter_bounds(self):
+        """
+        Inferred numerical bounds on the uncertainty set.
+        """
+        nom_val = self.origin
+        psi_mat = self.psi_mat
+
+        F = self.number_of_factors
+        beta_F = self.beta * F
+        floor_beta_F = math.floor(beta_F)
+        parameter_bounds = []
+        for i in range(len(nom_val)):
+            non_decreasing_factor_row = sorted(psi_mat[i], reverse=True)
+            # deviation = sum_j=1^floor(beta F) {psi_if_j} + (beta F - floor(beta F)) psi_{if_{betaF +1}}
+            # because indexing starts at 0, we adjust the limit on the sum and the final factor contribution
+            if beta_F - floor_beta_F == 0:
+                deviation = sum(non_decreasing_factor_row[j] for j in range(floor_beta_F - 1))
+            else:
+                deviation = sum(non_decreasing_factor_row[j] for j in range(floor_beta_F - 1)) + (
+                            beta_F - floor_beta_F) * psi_mat[i][floor_beta_F]
+            lb = nom_val[i] - deviation
+            ub = nom_val[i] + deviation
+            if lb > ub:
```

Comment:
Should this test have a tolerance?

Reply:
I'm not sure...is it recommended for comparing magnitudes of floating point values in Python?

Reply:
I don't have a good answer.  In general, I would have written it as you did, but having just spent time working with @michaelbynum on FBBT, his tests for "`lb > ub`" all include a tolerance.  I suppose the question is if this test is included as a "defensive programming" measure, or if you think it is likely that lb will be very close to (or EPS greater than) ub.

Reply:
It's mostly a defensive programming measure in this case.  I think the restriction we have on the Psi matrix for the factor model sets having all entries >= 0 ensures that lb cannot be significantly greater than ub, if at all. 

On that note, is possible that someone defines an uncertainty set in a given uncertain parameter set where lb == ub for that parameter, meaning it collapses to a single value in the set and it no longer needs to be treated as uncertain. I have notes on handling cases like this as future work, but for now we don't have any efficiencies for this case.